### ISSUE TITLE
Not sure why AUCMeter cannot be used if batch_size equals 1?

### ISSUE THREAD

**Issue by threefoldo** [catalyst-team/catalyst#898]
If the AUCMeter.add accepts a list of 1 item for binary classification, the following code will convert it to a number, not a list, then the assertion after this conversion will be triggered. 

Why is this conversion necessary?

```
from catalyst.tools.meters import AUCMeter

meter = AUCMeter()
probabilities = torch.tensor([[0.8281, 0.1719]])
targets = torch.tensor([[1., 0.]])
for i in range(2):
    meter.add(probabilities[:, i], targets[:, i])
```

https://github.com/catalyst-team/catalyst/blob/508e63039b5fe830afba5280fafc31fa6007b6b0/catalyst/tools/meters/aucmeter.py#L49-L56

**Comment by Scitator** [comment]
Hi, generally speaking for AUCCallback we expect the input like
```python
        outputs (torch.Tensor): NxK tensor that for each of the N examples
            indicates the probability of the example belonging to each of
            the K classes, according to the model.
        targets (torch.Tensor):  binary NxK tensort that encodes which of the K
            classes are associated with the N-th input
            (eg: a row [0, 1, 0, 1] indicates that the example is
            associated with classes 2 and 4)
```

current AUCMeter works **only** for binary classification problem with `outputs: Tensor [bs; ]` and `targets: Tensor [bs; ]`.

Moreover, we are working on metric standardisation now:
- issue: https://github.com/catalyst-team/catalyst/issues/786
- PR: https://github.com/catalyst-team/catalyst/pull/897
we would be glad to see your thoughts on it or you can join our slack and discuss it with the team - https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw

**Comment by threefoldo** [comment]
Thanks, I have joined the slack channel.
